### ISSUE TITLE
scylla_selinux_setup set the wrong value

### ISSUE THREAD

**Issue by ebenzecri** [scylladb/scylladb#9149]
If you run `scylla_selinux_setup` it will write the file `/etc/selinux/config`, setting `SELINUX` as `"disabled"` instead of `disabled`.

The original file content is:
```
# This file controls the state of SELinux on the system.
# SELINUX= can take one of these three values:
#     enforcing - SELinux security policy is enforced.
#     permissive - SELinux prints warnings instead of enforcing.
#     disabled - No SELinux policy is loaded.
SELINUX=enforcing
# SELINUXTYPE= can take one of these three values:
#     targeted - Targeted processes are protected,
#     minimum - Modification of targeted policy. Only selected processes are protected.
#     mls - Multi Level Security protection.
SELINUXTYPE=targeted
```

After scylla_selinux_setup was ran:
```
# This file controls the state of SELinux on the system.
# SELINUX= can take one of these three values:
#     enforcing - SELinux security policy is enforced.
#     permissive - SELinux prints warnings instead of enforcing.
#     disabled - No SELinux policy is loaded.
SELINUX="disabled"
# SELINUXTYPE= can take one of these three values:
#     targeted - Targeted processes are protected,
#     minimum - Modification of targeted policy. Only selected processes are protected.
#     mls - Multi Level Security protection.
SELINUXTYPE=targeted
```

It seems the problem is related to https://github.com/scylladb/scylla/blob/master/dist/common/scripts/scylla_util.py#L1004, specifically to `__escape()` function (and posible `set()` too).

Thanks!

**Comment by ebenzecri** [comment]
@syuu1228 Could you please take a look? Thanks!

**Comment by syuu1228** [comment]
This is not "wrong", because both are valid for sysconfig file, but it's better to follow distribution way to set the parameter.
I will send a PR for it

**Comment by ebenzecri** [comment]
@syuu1228 Was this finally fixed?

I did a local experiment today (because a customer asked about this a few days ago), setting `SELINUX="disabled"`, restarting the node and check SELinux status and unfortunately it was enabled, so setting SELINUX as **"disabled"** won't disable it at all (it actually set the mode as `permissive`).

Let me show you some evidence:
### SELINUX="disabled"
```
[scylla@centos70 ~]$ cat /etc/selinux/config 
# This file controls the state of SELinux on the system.
# SELINUX= can take one of these three values:
#     enforcing - SELinux security policy is enforced.
#     permissive - SELinux prints warnings instead of enforcing.
#     disabled - No SELinux policy is loaded.
SELINUX="disabled"
# SELINUXTYPE= can take one of three values:
#     targeted - Targeted processes are protected,
#     minimum - Modification of targeted policy. Only selected processes are protected. 
#     mls - Multi Level Security protection.
SELINUXTYPE=targeted 

[scylla@centos70 ~]$ sestatus 
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   permissive
Mode from config file:          error (Success)
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Max kernel policy version:      31

[scylla@centos70 ~]$ uptime
 20:11:23 up 0 min,  2 users,  load average: 1.37, 0.41, 0.14
[scylla@centos70 ~]$ 
```
### SELINUX=disabled
```
[scylla@centos70 ~]$ sestatus 
SELinux status:                 disabled

[scylla@centos70 ~]$ cat /etc/selinux/config 
# This file controls the state of SELinux on the system.
# SELINUX= can take one of these three values:
#     enforcing - SELinux security policy is enforced.
#     permissive - SELinux prints warnings instead of enforcing.
#     disabled - No SELinux policy is loaded.
SELINUX=disabled
# SELINUXTYPE= can take one of three values:
#     targeted - Targeted processes are protected,
#     minimum - Modification of targeted policy. Only selected processes are protected. 
#     mls - Multi Level Security protection.
SELINUXTYPE=targeted 

[scylla@centos70 ~]$ sestatus 
SELinux status:                 disabled
[scylla@centos70 ~]$ uptime
 20:14:05 up 0 min,  2 users,  load average: 1.16, 0.41, 0.14

```

Thanks!

**Comment by tomer-sandler** [comment]
@syuu1228 was this a regression introduced in Ent 2021.1, or was it not working as expected longer than that?
We still have many customers running 2020.1 (and even some with 2019.1) using CentOS / RHEL.

**Comment by ebenzecri** [comment]
Adding more info:

```
[user@xxxxxxxxxx ~]$ scylla --version
2020.1.7-0.20210304.e42189e09

[user@xxxxxxxxxx ~]$ sestatus 
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   permissive
Mode from config file:          error (Success)
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Max kernel policy version:      32

[user@xxxxxxxxxx~]$ cat /etc/selinux/config 
# This file controls the state of SELinux on the system.
# SELINUX= can take one of these three values:
#     enforcing - SELinux security policy is enforced.
#     permissive - SELinux prints warnings instead of enforcing.
#     disabled - No SELinux policy is loaded.
SELINUX="disabled"
# SELINUXTYPE= can take one of three values:
#     targeted - Targeted processes are protected,
#     minimum - Modification of targeted policy. Only selected processes are protected. 
#     mls - Multi Level Security protection.
SELINUXTYPE=targeted
```
```
[user@xxxxxxxxxx ~]$ scylla --version
2019.1.5-0.20200224.9fd24a74f

[user@xxxxxxxxxx ~]$ sestatus
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   permissive
Mode from config file:          error (Success)
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Max kernel policy version:      31

[user@xxxxxxxxxx ~]$ cat /etc/selinux/config 
# This file controls the state of SELinux on the system.
# SELINUX= can take one of these three values:
#     enforcing - SELinux security policy is enforced.
#     permissive - SELinux prints warnings instead of enforcing.
#     disabled - No SELinux policy is loaded.
SELINUX="disabled"
# SELINUXTYPE= can take one of three two values:
#     targeted - Targeted processes are protected,
#     minimum - Modification of targeted policy. Only selected processes are protected. 
#     mls - Multi Level Security protection.
SELINUXTYPE=targeted
```

**Comment by vladzcloudius** [comment]
@syuu1228 Did you see @ebenzecri comment (above https://github.com/scylladb/scylladb/issues/9149#issuecomment-988336043)? 

Based on Eduardo's experiment there is a big difference between unquoted and quoted value. And this seems to be a regression, @tomer-sandler.

The issue is more than 1 year old and it's quite about time it's fixed.


@DoronArazii FYI.
Please, prioritize.

**Comment by syuu1228** [comment]
There was a long time pending PR to fix this, stalled middle of review:
https://github.com/scylladb/scylladb/pull/9153

I will try to merge it soon.
@vladzcloudius could you review it please?

**Comment by vladzcloudius** [comment]
> There was a long time pending PR to fix this, stalled middle of review: #9153
> 
> I will try to merge it soon. @vladzcloudius could you review it please?

Done, @syuu1228. 
Please, see Eduardo's report above (I referenced it in my previous comment too).

**Comment by mykaul** [comment]
@syuu1228 - do we need a backport to 5.1 as well?
